test/sequential/test-deprecation-flags.js

```diff
@@ -32,7 +60,170 @@ execFile(node, traceDep, function(er, stdout, stderr) {
+    'directory. This functionality is deprecated and will be removed soon.',
+    ''
+  ];
+
+  assert.deepEqual(lines, messages);
+
+  console.log('normal ok');
+});
+
+execFile(node, ['--trace-deprecation', depprint], function(e, stdout, stderr) {
+  console.error('--trace-deprecation: testing internal/util\'s ' +
+                'printDeprecationMessage');
+  assert.equal(e, null);
+  assert.equal(stdout, '');
+  var lines = stderr.split('\n');
+  assert.equal(lines.length, 45);
+
+  assert.equal(lines[0],
+    'Trace: (node) smalloc is deprecated. Use typed arrays instead.');
+  assert.equal(lines[11],
+    'Trace: (node) sys is deprecated. Use util instead.');
+  assert.equal(lines[22],
+    'Trace: (node) Buffer.write(string, encoding, offset, length) is ' +
+    'deprecated. Use write(string[, offset[, length]][, encoding]) instead.');
+  assert.equal(lines[33],
+    'Trace: (node) warning: require(\'.\') resolved outside the package ' +
+    'directory. This functionality is deprecated and will be removed soon.');
+  assert.equal(lines[44], '');
+
+  console.log('trace ok');
+});
+
+execFile(node, [depAll], function(er, stdout, stderr) {
+  console.error('normal: testing all the deprecated functions/properties');
+  assert.equal(er, null);
+  assert.equal(stdout, '');
+  const lines = stderr.split('\n');
+  const messages = [
+    'http.Client is deprecated.',
+    'http.createClient is deprecated. Use http.request instead.',
+    'child_process: options.customFds option is deprecated. ' +
+    'Use options.stdio instead.',
+    'tty.setRawMode is deprecated. Use process.stdin.setRawMode instead.',
+    'readline.codePointAt is deprecated. ' +
+    'Use String.prototype.codePointAt instead',
+    'OutgoingMessage.flush is deprecated. Use flushHeaders instead.',
+    'Server.connections property is deprecated. ' +
+    'Use Server.getConnections method instead.',
+    'Server.connections property is deprecated.',
+    'Server.listenFD is deprecated. ' +
+    'Use Server.listen({fd: <number>}) instead.',
+    'os.getNetworkInterfaces is deprecated. Use os.networkInterfaces instead.',
+    '_writableState.buffer is deprecated. ' +
+    'Use _writableState.getBuffer instead.',
+    'crypto.createCredentials is deprecated. ' +
+    'Use tls.createSecureContext instead.',
+    'crypto.Credentials is deprecated. Use tls.createSecureContext instead.',
+    'Buffer.get is deprecated. Use array indexes instead.',
+    'Buffer.set is deprecated. Use array indexes instead.',
+    'util.p is deprecated. Use console.error instead.',
+    'util.exec is deprecated. Use child_process.exec instead.',
+    'util.print is deprecated. Use console.log instead.',
+    'util.puts is deprecated. Use console.log instead.',
+    'util.error is deprecated. Use console.error instead.',
+    'util.pump is deprecated. Use readableStream.pipe instead.',
+    'util.debug is deprecated. Use console.error instead.',
+    'DEBUG: undefined',
+    ''
+  ];
```

Comment:
Is it me or is this getting a bit out of hand? I feel testing for exact strings is a bit too much, as every deprecation change would require a test change as well. I mean it's nice to test thoroughly, but ultimately it's unnecessary overhead imo.